images/__common.sh

```diff
@@ -12,12 +12,13 @@ function downloadFileAndVerifyChecksum() {
     local downloadedFileName="$3"
     local downloadableFileName="$3"
     local sdkStorageAccountUrl="$4"
+    local sasToken="$5"
     local headersFile="/tmp/headers.txt"
 
     echo "Downloading $platformName version '$version' from '$sdkStorageAccountUrl'..."
     request="curl 
         -D $headersFile 
-        -SL $sdkStorageAccountUrl/$platformName/$downloadableFileName 
+        -SL $sdkStorageAccountUrl/$platformName/$downloadableFileName$sasToken
         --output $downloadedFileName"
     $__CURRENT_DIR/retry.sh "$request"
```

Comment:
`retry.sh` here has an `echo` whenever it fails to download, which could expose the `sasToken` when running in our pipelines. We should make sure that doesn't happen by removing the echo or adding some logic to only echo it in some cases

Reply:
That's a great point Paul!

Reply:
@waliMSFT did we get an answer for this?

Reply:
+1 to the above point

Reply:
Re-opening. Sorry having the token in the URL still makes me a bit concerned about the secrets showing up in the logs later since it's part of the URL. Like what happens if the base URL is ever wrong, will the SassToken still appear as `***` I image the SassToken eventually has to decrypt before storing in the string. 

Have we considered passing this as a header? Example:
curl -D headers.txt -SL "{sdkStorageBaseUrl}/{platformName}/{platformName}-{version}.tar.gz" --header "Authorization: Bearer $ORYX_SDK_STORAGE_ACCOUNT_ACCESS_TOKEN"

Reply:
Throughout the build script we are printing out the token. 

This can be seen when you run `export BUILDKIT_PROGRESS=plain` in git bash before executing the `buildBuildImages.sh` script. This will make sure that the full log output doesn't get hidden in your git bash terminal, and you can see what the actual output is, similar to how the full output of `docker build` commands are printed in the Azure Pipeline.

The token primarily is printed due to the `set -ex` found throughout all of the Oryx script that are run during the docker builds.

Technically, it isn't getting leaked since the Azure Pipeline is masking the token, but relying on this functionality does not seem (to me) like good practice.

Reply:
@william-msft @pauld-msft  masking the SasToken does not depend on the URL. As we are retrieving the KeyVault secret from the azure library, Azure considers it as secret and does not expose the secret. I thought this gives us the efficient way of handling the secrets and Azure is taking care of it. 
In the local builds we do not retrieve the token from keyVault if the account url does not point to the Staging one. 
Besides, in the local builds and the tests when the sasToken is retrieved from the KeyVault I avoided printing the steps and  the token in the logs by adding `+x` and `-x`  at the beginning and end respectively of the method.
https://github.com/microsoft/Oryx/pull/1878/files#:~:text=function%20retrieveSastokenFromKeyvault(),%7D

Reply:
@waliMSFT and I caught up on this offline, and he has addressed the issue with [this commit](https://github.com/microsoft/Oryx/pull/1878/commits/4ff61b4b3c591e08fb895c344fee2090999001f3) that disables the debugging flag `-x` wherever we are using this token.

Reply:
nit: would be nice if we could toggle this on via flag or something, to help us debug in the future.

Reply:
@william-msft thanks for the suggestion. For now, I will keep it as it is. an

> > > > * [ ]     As @cormacpayne mentioned in the meeting, lets add tests to make sure we don't propagate the token in any of the images
> > > 
> > > 
> > > @pauld-msft I have added a work item regarding this. https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1818074 @daniv-msft
> > 
> > 
> > @waliMSFT - so these tests are supposed to be an automated way of ensuring that the token is secure (at least in the final images). Until we have these in place, or some manual verification, I'm not sure that we would be able to be certain the token is not exposed. So is the idea to just consider the storage account as public until then, or are you planning on manually verifying that they don't get passed through?
> 
> +1 @waliMSFT could we at least provide some automated tests that will test the core functionality?
> 
> Do we have a testing plan for this? Either in the description or in a linked document will help us identify gaps in our testing plan before releasing to production. This is especially important if don't rely on any automated tests. Debugging this work in production in the midst of an expected pre-release will not be fun.

We discussed this issue recently and as we discussed I believe we do not need additional tests as functionality-wise or workflow-wise nothing changed. The main changes in this PR is the staging-url and passing the token with the URL to access the resources. For this, we have the tests to check if the account is working fine or not. 
https://github.com/microsoft/Oryx/blob/6c59cde0b3e22ecbdff11b1a9a194cb8a1b61a9d/tests/Oryx.Integration.Tests/StagingStorageAccountSanityTest.cs

Here I have just pointed to the new Staging account to test.

Reply:
> nit: would be nice if we could toggle this on via flag or something, to help us debug in the future.

Hi William, thanks for the suggestion. I have created a work item:
https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1831398